gradle-plugin/src/test/java/org/dmfs/gradle/gver/tasks/TagReleaseTaskTest.java

```diff
@@ -1,159 +1,98 @@
-                                ((GitVersionConfig) p.getExtensions().getByName("gver")).mChangeTypeStrategy = new Strategy();
-                                ((GitVersionConfig) p.getExtensions().getByName("gver")).mChangeTypeStrategy.mChangeTypeStrategies.addAll(
-                                    asList(
-                                        MAJOR.when(new CommitMessage(new Contains("#major"))),
-                                        MINOR.when(new CommitMessage(new Contains("#minor"))),
-                                        PATCH.when(new CommitMessage(new Contains("#patch"))),
-                                        UNKNOWN.when(((repository1, commit, branches) -> true))));
-                                return p;
-                            },
-                            project -> having(
-                                "p",
-                                proc -> repo -> proc.process(project),
-                                processes(() -> repository,
-                                    having(new Unchecked<Repository, Iterable<String>, Exception>(r -> new Mapped<>(Ref::getName, new Git(r).tagList().call())),
-                                        containsInAnyOrder(R_TAGS + "0.0.1", R_TAGS + "0.1.0"))
-                                ))))));
+    Assertion gitTagRelease_no_tag_on_feature_branch =
+        withResources("with temporary directory and git repository",
+            new TempDir(),
+            new Repository(getClass().getClassLoader().getResource("0.2.0-alpha.1.feature.bundle"), "feature"),
 
-    }
+            (tempDir, repo) -> withResource(new TestProject(tempDir,
+                    new Strategy(
+                        MAJOR.when(new CommitMessage(new Contains("#major"))),
+                        MINOR.when(new CommitMessage(new Contains("#minor"))),
+                        PATCH.when(new CommitMessage(new Contains("#patch"))),
+                        UNKNOWN.when(((repository1, commit, branches) -> true)))),
+                project -> assertionThat(
+                    repository -> {
+                        try
+                        {
+                            ((TagReleaseTask) project.getTasks().getByName("gitTagRelease")).perform();
+                            fail("did not throw");
+                        }
+                        catch (Exception e)
+                        {
+                            // ignore the exception until we have a Confidence Quality to test for it
+                            // for now we just ensure no tag has been added
+                        }
+                    },
+                    is(procedureThatAffects(
+                        new Text("alters repository"),
+                        () -> repo,
+                        soIt(has(repository -> new Mapped<>(Ref::getName, new Git(repository).tagList().call()),
+                            iteratesInAnyOrder(R_TAGS + "0.0.1", R_TAGS + "0.1.0"))))))));
```

Comment:
@sschuberth you probably can use this test case as the basis for your test. It verifies that no tags have been created when the user tries to create a release tag on a feature branch.